trackers/react-native-tracker/test/plugins/app_lifecycle.test.ts

```diff
@@ -0,0 +1,115 @@
+import { AppState } from 'react-native';
+import { BACKGROUND_EVENT_SCHEMA, LIFECYCLE_CONTEXT_SCHEMA } from '../../src/constants';
+import { buildPageView, Payload, trackerCore } from '@snowplow/tracker-core';
+import { newAppLifecyclePlugin } from '../../src/plugins/app_lifecycle';
+
+describe('Application lifecycle plugin', () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('tracks events on app state changes', async () => {
+    const appStateSpy = jest.spyOn(AppState, 'addEventListener');
+
+    const payloads: Payload[] = [];
+    const tracker = trackerCore({
+      callback: (pb) => payloads.push(pb.build()),
+      base64: false,
+    });
+    const appLifecyclePlugin = await newAppLifecyclePlugin(
+      {
+        appBuild: '19',
```

Comment:
what are these up to here?

Reply:
Good point, these are not relevant here as we are not testing the application context entity but the lifecycle events. I will remove them!